fence/agents/scsi/fence_scsi.py

```diff
@@ -182,7 +182,14 @@ def get_cluster_id(options):
 	cmd = options["--corosync-cmap-path"] + " totem.cluster_name"
 
 	match = re.search(r"\(str\) = (\S+)\n", run_cmd(options, cmd)["out"])
-	return hashlib.md5(match.group(1)).hexdigest() if match else fail_usage("Failed: cannot get cluster name")
+
+	if not match:
+		fail_usage("Failed: cannot get cluster name")
+
+	try:
+		return hashlib.md5(match.group(1), usedforsecurity=False).hexdigest()
+	except TypeError:
+		return hashlib.md5(match.group(1)).hexdigest()
```

Comment:
Fedora/RHEL/CentOS is a huge majority of the users so I don't see an issue to adapt to their needs. There are already several decisions that are not compatible with other distributions and you have to fine-tune it a bit if required. E.g. depending on the tools that are not packaged or do not work with latest upstream versions.